crates/ruff_python_resolver/src/native_module.rs

```diff
@@ -8,7 +8,26 @@ pub(crate) fn is_native_module_file_extension(file_extension: &OsStr) -> bool {
     file_extension == "so" || file_extension == "pyd" || file_extension == "dylib"
 }
 
-/// Returns `true` if the given file name is that of a native module.
-pub(crate) fn is_native_module_file_name(_module_name: &Path, _file_name: &Path) -> bool {
-    todo!()
+/// Given a file name, returns the name of the native module it represents.
+///
+/// For example, given `foo.abi3.so`, return `foo`.
+pub(crate) fn native_module_name(file_name: &Path) -> Option<&str> {
+    file_name
+        .file_stem()
+        .and_then(OsStr::to_str)
+        .and_then(|file_stem| file_stem.split_once('.').map(|(file_stem, _)| file_stem))
+}
+
+/// Returns `true` if the given file name is that of a native module with the given name.
+pub(crate) fn is_native_module_file_name(module_name: &str, file_name: &Path) -> bool {
```

Comment:
I'd add the following test cases:
* `foo.abi3.so`
* `foo.cpython-38-x86_64-linux-gnu.so
* `foo.cp39-win_amd64.pyd`

It doesn't seem to support `foo.so`, which iirc also works (but you shouldn't use when shipping)

negative:
* `libopenblasp-r0-41284840.3.18.s`

Reply:
Thank you, good call.